python-scripts/xbox/interface/if_xbdm.py

```diff
@@ -77,21 +77,30 @@ def xbdm_command(cmd, length=None):
   return lines
 
 def GetModules():
-  return [] # FIXME: Make this work..
+  modulesList = {}
   lines = xbdm_command("modules")
-  for line in lines:
-    print(line)
-    chunks = line.split()
-    for chunk in chunks:
-      if (chunk[0:6] == b'name=\"'):
-        print("  name = '" + str(chunk[6:-1]) + "'")
-      elif (chunk[0:5] == b'base='):
-        print("  base = '" + str(chunk[5]) + "'")
-      else:
-        print('  unparsed: ' + str(chunk))
-        pass
-        #assert(False)
-  return []
+  lines = lines.decode("utf-8") 
+
+  linesList = lines.splitlines() # if we get multiline response -> #name="xboxkrnl.exe" #name="xbdm.dll" etc
+
+  for line in linesList: 
+
+    nameAddrDict ={}
+
+    nameAddrChunks = line.split()
+
+    for nameAddrChunk in nameAddrChunks:
+       singleChunk = nameAddrChunk.split("=")    #name="xbdm.dll" ->  ['name', '"xbdm.dll"']
```

Comment:
Not every key will contain a value, see the example for ` #name="wavebank.exe" base=0x00010c40 size=0x0004fa40 check=0x00000000 timestamp=0x4c14abd2 tls xbe` which contains `tls` and `xbe` without value